Add email-verification-related records for support.digitalgov.gov

noremmie opened the pull request:
PRs affecting a Federalist site must receive approval from a member of the relevant team.

The Search.gov team is transitioning from sending emails via Mandrill to sending emails via AWS SES.

This PR adds four new `support.digitalgov.gov` DNS records that are needed to set up AWS SES to send emails on behalf of the [search-gov app](https://github.com/GSA/search-gov). It also modifies an existing SPF record. We intend to remove Mandrill-related content from DNS records after the transition is complete.

This summarizes the four records I'm attempting to create.

| Purpose                 | DNS Record Name                                                      | DNS Record Type | DNS Record Value                                      |
| ----------------------- | -------------------------------------------------------------------- | --------------- | ----------------------------------------------------- |
| SES Domain Verification | `_amazonses.support.digitalgov.gov.`                                  | `TXT`           | `T5etn/YylzSUQQWw6HspyK4+2+B9XzE7Kajpz9ogfJI=`        |
| DKIM                    | `4ixtpnvpubjuxqvnex727otq55y2ew7w._domainkey.support.digitalgov.gov.` | `CNAME`         | `4ixtpnvpubjuxqvnex727otq55y2ew7w.dkim.amazonses.com` |
| DKIM                    | `tmoxp5vgftwsmhkukt2z6ayvfj5bw7zo._domainkey.support.digitalgov.gov.` | `CNAME`         | `tmoxp5vgftwsmhkukt2z6ayvfj5bw7zo.dkim.amazonses.com` |
| DKIM                    | `5uiojtkg7z5grkldq7ajm3zamtzh3h2s._domainkey.support.digitalgov.gov.` | `CNAME`         | `5uiojtkg7z5grkldq7ajm3zamtzh3h2s.dkim.amazonses.com` |

I've double checked my work, and I think the change is correct. But this is my first PR here, so a careful review would be appreciated. Thanks!

dawnpm commented:
@jeremyzilar here's our DNS update request. Could you give it a look? Thanks.

jeremyzilar commented:
@noremmie @dawnpm — did you check all the other entries in this file for `support.digitalgov.gov` to see that they are needed? There are quire a few.

Maybe you can also move them closer together a note which might be depreciated in the comments in the file?

noremmie commented:
@jeremyzilar I took a look at the other `support.digitalgov.gov` entries in that file, and I found five. We will be able to remove two of them once we're no longer using Mandrill. I added comments to them indicating as much:

* `k1._domainkey.support.digitalgov.gov.`
* `mandrill._domainkey.support.digitalgov.gov.`

I added those comments in a new commit that can be squashed when this PR is merged. If you'd rather I squash them together in advance, I can do that too.

The other three are A, MX, and TXT records all named `support.digitalgov.gov.`, and I don't know whether anyone outside of our team is relying on those records. I don't think it would be safe to remove those, or at least I don't think I have the authority to remove those.

I can't say whether any of the non-`support.digitalgov.gov` entries in that file are still needed (e.g., the `email.digitalgov.gov.` and `dzc.digitalgov.gov.` TXT records).

The four new records I'm requesting all need to be permanent or at least as long as we'll be using SES to send emails. I put the new records as close to each other as I could, and the three CNAME records are contiguous. The fourth is not grouped with them because it's a TXT record, and TXT records appear to have their own section in this file.

jeremyzilar commented:
@noremmie Great — This looks good. Keep me updated about it's merge.

noremmie commented:
Thanks, @erik-burgess. I see [the CircleCI failure resulting from this change](https://circleci.com/gh/18F/dns/301), but I don't understand what's wrong or different about the new `5uiojtkg7z5grkldq7ajm3zamtzh3h2s._domainkey.support.digitalgov.` record. The other two DKIM-related CNAMEs didn't result in any trouble. Please let me know if you want me to help troubleshoot anything.

noremmie commented:
Actually, I see the problem now, and it's a silly mistake on my part. I'll fix this with a follow-up PR.

noremmie commented:
@erik-burgess I've addressed the problem in a new pull request:

* https://github.com/18F/dns/pull/223

I apologize for the trouble. This is a silly mistake that I should have spotted beforehand.
